tools/r2d2/custom.py

```diff
@@ -0,0 +1,148 @@
+import pandas as pd
+import sys
+import time
+
+prediction_value = None
+
+
+class R2D2:
+    MEGA = 10 ** 6
+    MEGA_STR = " " * MEGA
+
+    CMD_COL = "cmd"
+    ARG_COL = "arg"
+
+    def __init__(self):
+        self._mem_array = []
+        print("R2D2 __init__")
+
+    def _mem_size_mb(self):
+        return len(self._mem_array)
+
+    def _clear_memory(self):
+        self._mem_array = []
+
+    def _alloc_additional_memory(self, additional_mb):
+        for i in range(0, additional_mb):
+            try:
+                self._mem_array.append(R2D2.MEGA_STR + str(i))
+                print("Adding: {}".format(i))
+            except MemoryError:
+                print("No more memory here..")
```

Comment:
More informative message. You could write: "Can not allocate more memory. Total allocated memory: XXX "

Reply:
sure.